slsim/Sources/source.py

```diff
@@ -68,20 +74,25 @@ def kwargs_variability_extracted(self):
             ##Here we prepare variability class on the basis of given
             # kwargs_variability.
             kwargs_variab_extracted = {}
-            kwargs_variability_list = ["supernovae_lightcurve"]
```

Comment:
I suggest to use another kwargs_variability_list for agn. Let's say kwargs_variability_list_agn = ["agn_lightcurve"]. Then, write elif statement for agn with out touching if statement. This will give us waht is happening here.

Reply:
Got it sorted! Apparently I added an indent to one of the loops.